### PR TITLE
add patterns: Singleton + Prototype; Proxy + Decorator; Observer + St…

### PR DESCRIPTION
**Pull request by kkapytau**
Please check task 4

### CODE REVIEW

File: src/flux/AppDispatcher.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Dispatcher } from 'flux';
```
Comment (yury-kuzmiankov) [inline src/flux/AppDispatcher.js]: in my opinion flux should be implemented without any libs

**Review by yury-kuzmiankov — COMMENTED** [review]
Please, put show me examples of Prototype, Decorator

File: src/flux/AppDispatcher.js (hunk 1 above)
Comment (kkapytau) [reply src/flux/AppDispatcher.js]: Hi Yury, Prototype - it is App.js file (NewsApp function). Decorator it is NewsItem.js file -> ItemDecorator function

File: src/flux/AppDispatcher.js [hunk 2]
```diff
@@ -1,4 +1,23 @@
-import { Dispatcher } from 'flux';
+//import { Dispatcher } from 'flux';
```
Comment (yury-kuzmiankov) [inline src/flux/AppDispatcher.js]: i'm ok with new implementation. Is there decorator here? 'handleAction' ? Usually decorator replace/extend part of current functionality with new one

Comment (kkapytau) [reply src/flux/AppDispatcher.js]: It is just common method for Dispatcher instance. We can void using 'handleAction' and instead of it use - 'dispatch'

**Pull request merged**